Main.java

```diff
@@ -1,237 +1,448 @@
+
 import javafx.application.Application;
+import javafx.beans.property.SimpleDoubleProperty;
 import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
 import javafx.collections.ObservableList;
 import javafx.geometry.Insets;
 import javafx.geometry.Pos;
 import javafx.scene.Scene;
 import javafx.scene.control.Button;
-import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
 import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableColumn.SortType;
 import javafx.scene.control.TableView;
-import javafx.scene.control.TextArea;
 import javafx.scene.control.TextField;
 import javafx.scene.control.cell.PropertyValueFactory;
 import javafx.scene.layout.GridPane;
 import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.Region;
 import javafx.scene.layout.VBox;
 import javafx.scene.text.Text;
 import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
 import javafx.stage.Stage;
 
 /**
-* Entry point of the FoodQuery Program
-* Houses GUI code for now.
-*/
+ * Entry point of the FoodQuery Program
+ * Houses GUI code for now.
+ */
 public class Main extends Application {
 
+	// Application's stage
+	Stage mainStage;
+	// Stage's Scene
+	Scene mainScene;
+	// Scene's overarching grid
+	GridPane grid;
+
+	// Application's buttons
+	Button loadFileButton, saveFileButton, newFoodButton, clearFoodEntry, addToMenu, removeFromMenu, nameQueryButton, nutriQueryButton, clearQueries, analyzeMeal;
+
+	// Applications Containers
+	VBox transferButtons, queries;
+	HBox listMenu, nameQuery, nutriQuery, foodItemInsert;
+
+	// Application's text fields
+	TextField insertId, insertName, insertCals, insertCarbs, insertFat, insertProtein, insertFiber, nameText, queryAmount, counter;
+	Text orText;
+
+	// Applications data structures
+	// Underlying foodlist object
+	FoodData mainFoodList;
+	// current viewable food list as it's being altered by filters or additions
+	ObservableList<FoodItem> tempFoodList;
+	// the list used in the menu
+	ObservableList<FoodItem> menuList;
+
+	TableView<FoodItem> foodListTable, menuListTable;
+
+
 	public static void main(String[] args) {
 		launch(args);
 	}
 
 	@Override
 	public void start(Stage primaryStage){
-		Stage mainStage = new Stage();
+		// MainStage Settings
+		mainStage = new Stage();
 		mainStage.setTitle("Menu Selector");
-		
-		GridPane grid = new GridPane();
-//		grid.setGridLinesVisible(true);
+		mainStage.setMaximized(false);
+
+		// Grid Settings
+		grid = new GridPane();
+		grid.setGridLinesVisible(false);
 		grid.setAlignment(Pos.CENTER);
 		grid.setVgap(5);
 		grid.setHgap(5);
 		grid.setPadding(new Insets(10,10,10,10));
+
+
+
+		mainFoodList = new FoodData();
+		mainFoodList.loadFoodItems("foodItems.txt");
+
+		tempFoodList = FXCollections.observableArrayList(mainFoodList.getAllFoodItems());
+//        tempFoodList.addListener(new ListChangeListener<FoodItem>() {

```

Comment:
What is this commented out section for? Can you get rid of it?

Reply:
Commented section is work-in-progress code so tables will update automatically when changes are made like querying.